modules/query-performance-analysis.adoc

```diff
@@ -0,0 +1,50 @@
+:_newdoc-version: 2.18.4
+:_template-generated: 2025-09-22
+:_mod-docs-content-type: CONCEPT
+
+[id="query-performance-analysis_{context}"]
+= Query performance analysis
+
+Every query and subquery in Loki generates a `metrics.go` log line with performance statistics. Subqueries emit the log line in the queriers. Every query has an associated single summary `metrics.go` line emitted by the query-front end.
+
+Look for the following values in the statistics:
+
+* `duration`: Denotes the amount of time a query took to run.
+* `queue_time`: Denotes the time a query spent in the queue before being processed.
+* `chunk_refs_fetch_time`: Denotes the amount of time spent in getting chunk information from the index.
+* `store_chunks_download_time`: Denotes the amount of time in getting chunks from cache or storage.
+
```

Comment:
You have to add the following calculation otherwise this doesn't make sent

total_duration = duration + queue_time

Queue Time = queue_time / total_duration * 100
Chunk Refs Fetch Time = chunk_refs_fetch_time / total_duration * 100
Chunks Download Time = store_chunks_download_time / total_duration * 100
Execution Time = (duration - chunk_refs_fetch_time - store_chunks_download_time) / total_duration * 100